answerking_app/tests/test_unit/test_unit_retiremixin.py

```diff
@@ -0,0 +1,93 @@
+from enum import auto
+from unittest import mock
+
+from rest_framework.request import Request
+
+from answerking_app.models.models import Category, Product, Tag
+from answerking_app.tests.test_unit.UnitTestBaseClass import UnitTestBase
+from answerking_app.utils.mixins.ApiExceptions import ProblemDetails
+from answerking_app.utils.mixins.RetireMixin import RetireMixin
+
+
+class RetireMixinUnitTests(UnitTestBase):
+    UTB = UnitTestBase()
+
+    mixin_path: str = "answerking_app.utils.mixins.RetireMixin"
+    cat_data = UTB.get_fixture("categories", "burgers_cat_data.json")
+    retired_cat_data = UTB.get_fixture("categories", "retired_cat_data.json")
+    product_data = UTB.get_fixture("products", "margarita_pizza_data.json")
+    retired_product_data = UTB.get_fixture(
+        "products", "retired_product_data.json"
+    )
+    cat: Category
+    retired_cat: Category
+    product: Product
+    retired_product: Product
+
+    def setUp(self):
+        Category.objects.create(**self.cat_data)
+        self.cat = Category.objects.get(name="Burgers")
+        Category.objects.create(**self.retired_cat_data)
+        self.retired_cat_data = Category.objects.get(name="Old Burgers")
+        Product.objects.create(**self.product_data)
+        self.product = Product.objects.get(name="Margarita pizza")
+        Product.objects.create(**self.retired_product_data)
+        self.retired_product = Product.objects.get(name="Old pizza")
+
+    def test_retire_mixin_retired_category(self):
```

Comment:
you could use the @ddt and @data decorator on this test to test category, product and tag. It would look something like this
`
@ddt()
class RetireMixinUnitTests(UnitTestBase):
...
@data(self.retired_cat_data, self.retired_prod_data, self.retired_tag_data)
def test_retire_mixin_retired_category(self, retired_data):
with mock.patch.object(
...
return_value=self.retired_data
):
...
`

Reply:
without the slashes